Lab1/app/src/main/java/com/example/jgozal1/lab1/MainActivity.java

```diff
@@ -0,0 +1,107 @@
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.main);
+        updateUI();
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        getMenuInflater().inflate(R.menu.menu,menu);
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case R.id.action_add_item:
+                AlertDialog.Builder builder = new AlertDialog.Builder(this);
+                builder.setTitle("Add an item");
+                builder.setMessage("What would you like to buy?");
+                final EditText inputField = new EditText(this);
+                builder.setView(inputField);
+                builder.setPositiveButton("Add", new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialogInterface, int i) {
+                        String item = inputField.getText().toString();
+
+                        helper = new ItemDBHelper(MainActivity.this);
+                        SQLiteDatabase db = helper.getWritableDatabase();
+                        ContentValues values = new ContentValues();
+
+                        values.clear();
+                        values.put(ItemContract.Columns.ITEM,item);
+
+                        db.insertWithOnConflict(ItemContract.TABLE,null,values,SQLiteDatabase.CONFLICT_IGNORE);
+                        updateUI();
+                    }
+                });
+
+                builder.setNegativeButton("Cancel",null);
+
+                builder.create().show();
+                return true;
+
+            default:
+                return false;
+        }
+    }
+
+    private void updateUI() {
+        helper = new ItemDBHelper(MainActivity.this);
+        SQLiteDatabase sqlDB = helper.getReadableDatabase();
+        Cursor cursor = sqlDB.query(ItemContract.TABLE,
+                new String[]{ItemContract.Columns._ID, ItemContract.Columns.ITEM},
+                null, null, null, null, null);
+
+        listAdapter = new SimpleCursorAdapter(
+                this,
+                R.layout.item_view,
+                cursor,
+                new String[]{ItemContract.Columns.ITEM},
+                new int[]{R.id.itemTextView},
+                0
+        );
+
+        this.setListAdapter(listAdapter);
+    }
+
+    public void onDoneButtonClick(View view) {
+        View v = (View) view.getParent();
+        TextView itemTextView = (TextView) v.findViewById(R.id.itemTextView);
+        String item = itemTextView.getText().toString();
+
+        String sql = String.format("DELETE FROM %s WHERE %s = '%s'",
+                ItemContract.TABLE,
+                ItemContract.Columns.ITEM,
+                item);
+
+
+        helper = new ItemDBHelper(MainActivity.this);
+        SQLiteDatabase sqlDB = helper.getWritableDatabase();
+        sqlDB.execSQL(sql);
+        updateUI();
+    }
```

Comment:
In the future, you should comment at the top of each function so others can read what the function does. No worries now, we didn't tell you